File: Controllers/Resources/SaveProductResource.cs
using System;
using System.Collections.Generic;
using pc_store.Core.Models;

namespace pc_store.Controllers.Resources
{
    public class SaveProductResource
    {
        public string Name { get; set; }
        public string Description { get; set; }
        public decimal Price { get; set; }
        public int Stock { get; set; }
        public int CategoryId { get; set; }
        public int SubCategoryId { get; set; }
        public ICollection<ProductPhoto> ProductPhotos { get; set; }
        public ICollection<SpecificationPhoto> SpecificationPhotos { get; set; }
        public string VideoLink { get; set; }
    }
}

File: ClientApp/app/shared/models/save-product-resource.ts
export interface SaveProductResource
{
    id: number;
    name: string;
    description: string;
    price: number;
    stock: number;
    productPhotoId?: number;
    specificationPhotoId?: number;
    categoryId: number;
    subCategoryId?: number | null;
    videoLink: string;
}